nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1152,20 +1154,54 @@ private Description handleInvocation(
       }
       nonNullPositions = builder.build();
     }
-    if (nonNullPositions.isEmpty()) {
-      return Description.NO_MATCH;
-    }
     // now actually check the arguments
+    ExpressionTree actual;
     for (int argPos : nonNullPositions) {
       // make sure we are passing a non-null value
-      ExpressionTree actual = actualParams.get(argPos);
+      actual = actualParams.get(argPos);
       if (mayBeNullExpr(state, actual)) {
         String message =
             "passing @Nullable parameter '" + actual.toString() + "' where @NonNull is required";
         return createErrorDescriptionForNullAssignment(
             MessageTypes.PASS_NULLABLE, actual, message, actual, state.getPath());
       }
     }
+    // Check for @NonNull being passed to castToNonNull (if configured)
```

Comment:
Can we extract this logic to its own function?

Reply:
Yup.